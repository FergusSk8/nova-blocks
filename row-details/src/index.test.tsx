import React from "react";
import { shallow } from "enzyme";
import RowDetails from './index';



const renderComponent = (props?: any) => {
  return shallow(<RowDetails {...props} />);
};

describe("Input", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should be create a snapshot", async   () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  describe('When exist numberOfSections', () => {

    const props = {
        numberOfSections: 4
    };

    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });

  describe("OnClickSection", () => {
    it("should call a action in a title section", () => {

    const component = renderComponent();
      component.find('p').at(0).simulate("click");
      component.update();
      expect(component).toMatchSnapshot();
    });

    it("should call a action in a content section", () => {

        const component = renderComponent();
          component.find('p').at(1).simulate("click");
          component.update();
          expect(component).toMatchSnapshot();
        });
  });
});
