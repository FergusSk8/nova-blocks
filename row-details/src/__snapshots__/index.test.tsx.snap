// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Fragment>
  <Row>
    <Col
      flex="200px"
    >
      <div />
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex="auto"
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Transfers and payments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              34
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex="auto"
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Taxes and contributions
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              33
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex="auto"
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Investments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              423
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex="auto"
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Paysheet
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              987
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex="auto"
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Administration
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              545
            </p>
          </div>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;

exports[`Input OnClickSection should call a action in a content section 1`] = `
<Fragment>
  <Row>
    <Col
      flex="200px"
    >
      <div />
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Transfers and payments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              34
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Taxes and contributions
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              33
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Investments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              423
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Paysheet
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              987
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Administration
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              545
            </p>
          </div>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;

exports[`Input OnClickSection should call a action in a title section 1`] = `
<Fragment>
  <Row>
    <Col
      flex="200px"
    >
      <div />
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Transfers and payments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              34
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Taxes and contributions
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              33
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Investments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              423
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Paysheet
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              987
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Administration
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              545
            </p>
          </div>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;

exports[`Input Should be create a snapshot 1`] = `
<Fragment>
  <Row>
    <Col
      flex="200px"
    >
      <div />
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Transfers and payments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              34
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Taxes and contributions
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              33
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Investments
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              423
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Paysheet
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              987
            </p>
          </div>
        </div>
      </div>
    </Col>
    <Divider
      type="vertical"
    />
    <Col
      flex={0.8}
    >
      <div>
        <div>
          <div>
            <p
              onClick={[Function]}
            >
              Administration
            </p>
          </div>
          <div>
            <p
              onClick={[Function]}
            >
              545
            </p>
          </div>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;
